### PR TITLE
[MRG+1] Remove ChunkedTransferMiddleware from default settings

### PR DESCRIPTION
**Pull request by redapple**
Following up on https://github.com/scrapy/scrapy/pull/1896

### CODE REVIEW

File: scrapy/downloadermiddlewares/chunked.py [hunk 1]
```diff
@@ -1,6 +1,14 @@
+import warnings
+
+from scrapy.exceptions import ScrapyDeprecationWarning
 from scrapy.utils.http import decode_chunked_transfer
 
 
+warnings.warn("Module `scrapy.downloadermiddlewares.chunked.ChunkedTransferMiddleware` "
```
Comment (redapple) [inline scrapy/downloadermiddlewares/chunked.py]: Updated. I used the module to make it shorter.

File: docs/topics/downloader-middleware.rst [hunk 2]
```diff
@@ -665,7 +665,13 @@ ChunkedTransferMiddleware
 
 .. class:: ChunkedTransferMiddleware
 
-   This middleware adds support for `chunked transfer encoding`_
+   This middleware adds support for `chunked transfer encoding`_.
+
+.. note::
+    This middleware is not enabled nor used by Scrapy downloader anymore.
+    In fact, Scrapy downloader has built-in support for chunked transfers,
+    so this middleware has no effect in practice.
```
Comment (kmike) [inline docs/topics/downloader-middleware.rst]: What do you think about removing docs for this middleware? It is deprecated, after all

Comment (redapple) [inline docs/topics/downloader-middleware.rst]: Yep. why not after all.

**Comment by kmike** [comment]
@redapple what do you think about putting this to 1.3?

**Comment by redapple** [comment]
@kmike , +1

**Pull request merged**